### ISSUE TITLE
Confusion in Validation or Testing

### ISSUE THREAD

**Issue by aliman80** [OpenGVLab/InternImage#77]
Hi , extremely obliged for sharing such an excellent and state of the art work. I have following queries if you can spare your precious time to answer please:
1:
in the upernet_internimage_b_512_160k_ade20k.py  following line is written at line no 61-63. 

data=dict(samples_per_gpu=2,
          val=dict(pipeline=test_pipeline),
          test=dict(pipeline=test_pipeline))

I have used my own dataset in the similar fomat and its giving me validation accuracy of 50.01 so my question is is this validation accuracy or testing accuracy?
How can i get the testing accuracy for it 

2: In my dataset i have 4 classes and one background class but in results i only see forground clases and no background classe , how can i add that .  I have tried to add background class in "CLASSES" list but then my prediction results give first four classes inclduding background and leave the last class as nan.

I would be very grateful if you can guide on these issues please

**Comment by czczup** [comment]
Hi, this is the config of `ADE20K` dataset.
```
# dataset settings
dataset_type = 'ADE20KDataset'
data_root = 'data/ADEChallengeData2016'
img_norm_cfg = dict(
    mean=[123.675, 116.28, 103.53], std=[58.395, 57.12, 57.375], to_rgb=True)
crop_size = (512, 512)
train_pipeline = [
    dict(type='LoadImageFromFile'),
    dict(type='LoadAnnotations', reduce_zero_label=True),
    dict(type='Resize', img_scale=(2048, 512), ratio_range=(0.5, 2.0)),
    dict(type='RandomCrop', crop_size=crop_size, cat_max_ratio=0.75),
    dict(type='RandomFlip', prob=0.5),
    dict(type='PhotoMetricDistortion'),
    dict(type='Normalize', **img_norm_cfg),
    dict(type='Pad', size=crop_size, pad_val=0, seg_pad_val=255),
    dict(type='DefaultFormatBundle'),
    dict(type='Collect', keys=['img', 'gt_semantic_seg']),
]
test_pipeline = [
    dict(type='LoadImageFromFile'),
    dict(
        type='MultiScaleFlipAug',
        img_scale=(2048, 512),
        # img_ratios=[0.5, 0.75, 1.0, 1.25, 1.5, 1.75],
        flip=False,
        transforms=[
            dict(type='Resize', keep_ratio=True),
            dict(type='RandomFlip'),
            dict(type='Normalize', **img_norm_cfg),
            dict(type='ImageToTensor', keys=['img']),
            dict(type='Collect', keys=['img']),
        ])
]
data = dict(
    samples_per_gpu=4,
    workers_per_gpu=4,
    train=dict(
        type=dataset_type,
        data_root=data_root,
        img_dir='images/training',
        ann_dir='annotations/training',
        pipeline=train_pipeline),
    val=dict(
        type=dataset_type,
        data_root=data_root,
        img_dir='images/validation',
        ann_dir='annotations/validation',
        pipeline=test_pipeline),
    test=dict( # please see here
        type=dataset_type,
        data_root=data_root,
        img_dir='images/validation',
        ann_dir='annotations/validation',
        pipeline=test_pipeline))
```
As shown, in this config, if you use `dist_test.py` to test the model, it will use validation set.

**Comment by czczup** [comment]
For example, if the labels of your custom dataset have the format of {0: background, 1: class1, 2: class2, 3: class3}, please set `reduce_zero_label=False` and `CLASSES=4`.

**Comment by czczup** [comment]
If possible, you can paste your config here, and I will provide further comments.

**Comment by aliman80** [comment]
Thank you so very much for your extemely helpful comments and guide. so what i understood is that the accuracy which i obtained is on the validation set and its validation accuracy . So to get the testing accuracy should i replace  test=dict(                                                         
        type=dataset_type,
        data_root=data_root,
        **img_dir='images/validation',   
        ann_dir='annotations/validation',**
        pipeline=test_pipeline))
as following 
 test=dict(                                                         
        type=dataset_type,
        data_root=data_root,
        **img_dir='images/test',   
        ann_dir='annotations/test',**
        pipeline=test_pipeline))
2. My second question is when i run the training script it the end it give my validation accuracy , inorder to get the test accuracy should i run the test.py 
3. In my case in annotations i dont have background mentioned rather i have this  {0:  "   " , 1: class1, 2: class2, 3: class3,4: class4}, if i keep reduce_zero_label=False and CLASSES=4. then i get very good improved results but no mentioned of background class.
4. Thank you again for your consistent support and time as this is helping me alot

**Comment by czczup** [comment]
1. Yes, that right.
2. Yes, after the first step you mentioned done, please run test.py to get test accuracy.
3. If you have {0: background , 1: class1, 2: class2, 3: class3,4: class4}, you should keep reduce_zero_label=False and CLASSES=5

**Comment by aliman80** [comment]
Thanks a lot
